corehq/apps/saved_reports/models.py

```diff
@@ -510,6 +541,16 @@ def datespan_filter_choices(datespan_filters, lang):
                 'slug': None,
             }] + localized_datespan_filters
 
+    def is_shared_on_domain(self):
+        # Is there a better way of checking this?
+        domain_scheduled_reports = ReportNotification.by_domain(self.domain, stale=False)
+        config_id = self._id
+
+        for report in domain_scheduled_reports:
+            if config_id in report.config_ids:
+                return True
+        return False
```

Comment:
Maybe
```suggestion
        config_ids = self.shared_on_domain(self.domain, only_id=True, stale=False)
        return self._id in config_ids
```
If `shared_on_domain()` were to leave `shared_config_ids` as a set if `only_id=True`, that would make this faster.